service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -337,8 +337,33 @@ echo "Setting up git integration..."
 # user starts a shell, we start the ssh-agent.
 cat << 'EOF' >> "${USER_BASH_PROFILE}"
 
-# Start the ssh-agent; "eval" it to set SSH_AUTH_SOCK and SSH_AGENT_PID
-eval "$(ssh-agent -s)"
+# Start a new ssh-agent if one is not already running.
```

Comment:
LGTM, but should this be moved to the "instance boot service" part so that it is run after reboot as well?

Reply:
That is an interesting thought, now that we have that service available.

I'd actually be interested in adding it as a standalone service that gets restarted if it crashes.
My experience with ssh-agent across the years is that I periodically find that it has stopped running, for reasons unclear, and I need to restart it myself. Doing this in the` .bash_profile` feels slightly wrong, but it does provide a level of control and simplicity for users.

I'm in the process of trying to figure out how the custom container on Vertex AI Workbench works. The answer to that might impact our long term decision. So I'm inclined to continue our current approach (just improved) and revisit soon,